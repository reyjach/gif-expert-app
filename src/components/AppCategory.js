import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AppCategory = ({ setCategories }) => {

  const [inputValue, setInputValue] = useState('');

  const hadleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const hadleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length >2){
      setCategories( cat => [inputValue,...cat]);
      setInputValue('');
    }
    
  }

  return (
      <form onSubmit={ hadleSubmit }>
        <input 
          type='text'
          value={inputValue}
          onChange={ hadleInputChange }
        />
      </form>
  )
}

AppCategory.propTypes = {
  setCategories:PropTypes.func.isRequired
}
