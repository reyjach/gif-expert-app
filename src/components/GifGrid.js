import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    const { data:images, loading }  = useFetchGifs( category );

    return (
        <>
            <h3 className='card animate__animated animate__bounceOutRight'>{category}</h3>
            {loading && <p className='card animate__animated animate__fadeOut'>Loading</p> }
            <div className='card-grid'>             
                { 
                    images.map(img => (
                    <GifGridItem key={img.id} {...img}/>
                    )) 
                }
            </div>
        </>
        
    )
}
GifGrid.propTypes = {
    category:PropTypes.string.isRequired
  }