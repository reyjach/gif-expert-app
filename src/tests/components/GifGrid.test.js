import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Prueba de GifGrid', () => { 

    const category = 'hola mundo';

    test('Debe de mostrarse correctamente', () => { 

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar item cuando se cargan imagenes useFetchGifs', () => { 

        const gifs = [{
            id: 'ABC',
            title: 'https://localhost/img.jpg',
            url: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);

        expect(wrapper).toMatchSnapshot()
    })
 })