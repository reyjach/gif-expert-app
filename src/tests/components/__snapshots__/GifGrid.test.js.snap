// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGrid Debe de mostrar item cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__bounceOutRight"
  >
    hola mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="https://localhost/img.jpg"
      url="Cualquier cosa"
    />
  </div>
</Fragment>
`;

exports[`Prueba de GifGrid Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="card animate__animated animate__bounceOutRight"
  >
    hola mundo
  </h3>
  <p
    className="card animate__animated animate__fadeOut"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
